### ISSUE TITLE
Make it so you can override or configure HttpService for HttpHealthIndicator

### ISSUE THREAD

**Issue by kevjames3** [nestjs/terminus#1151]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
Currently, if you need to override any of the Axios settings for `HttpHealthIndicator`, you must do that at call time and not injection time.  This is because the instance of `HttpHealthIndicator` has already been set.

## Expected behavior
I would like to be able to use a global/singleton instance of Axios/HTTPService that can be injected into Terminus, utilizing any configuration settings that have been set for Axios.  This would allow me to follow DRY principles.

## Minimal reproduction of the problem with instructions

`App.module.ts`
```ts
@Global()
@Module({
  imports: [
    TerminusModule,
    HttpModule.register({ // This does nothing, it has to occur in health.controller.ts instead
      httpsAgent: new https.Agent({
        rejectUnauthorized: false,
        cert: [fs.readFileSync(config.certificate)],
      }),
    }),
   // other modules...
  ],
  controllers: [HealthController],
  providers: [HttpService],
})
export class AppModule {
  constructor(private connection: Connection) {}

 // other stuff
}
```

`health.controller.ts`
```ts
@Controller('health')
export class HealthController {
  constructor(
    private health: HealthCheckService,
    private http: HttpHealthIndicator,
    private db: TypeOrmHealthIndicator,
  ) {}

  @Get()
  @HealthCheck()
  check() {
    return this.health.check([
      () =>
        this.http.pingCheck(
          'foo',
          'https://service-requiring-cert.com',
          {
            // This part must be set or Terminus will give "message": "self signed certificate in certificate chain"
            httpsAgent: new Agent({
              rejectUnauthorized: false,
              cert: [fs.readFileSync(config.certificate)],
            }),
          },
        ),
      () => this.db.pingCheck('database'),
    ]);
  }
}
```

## What is the motivation / use case for changing the behavior?
In my app, I need to make a call against a service that has a self-signed certificate (b/c healthchecks).  I currently set this certificate at the node runtime level, but instead of doing this globally using something like 'syswide-cas', I would rather use dependency injection that NodeJS provides.

## Environment

Nest version: 6.14.9

 
For Tooling issues:
- Node version: XX  14.15.3
- Platform:  Mac

**Comment by BrunnerLivio** [comment]
Sounds reasonable to me! Would you like to create a PR for this? :)

**Comment by kevjames3** [comment]
@BrunnerLivio - would love to, I just have no time at the moment.  Will loop around if I get to it though

**Comment by BrunnerLivio** [comment]
@kevjames3 I've added a new option; `httpClient`, where you can pass the `HttpService`-instance.

Try it via `npm i --save @nestjs/terminus@7.2.0-next.0`

Example:

`app.module.ts`

```typescript
@Module({
  imports: [
    HttpModule.register({ baseURL: 'https://google.com' }),
    TerminusModule,
  ],
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule {}
```

`app.controller.ts`

```typescript
@Controller()
export class AppController {
  constructor(
    private readonly http: HttpHealthIndicator,
    private readonly httpService: HttpService,
    private readonly health: HealthCheckService,
  ) {}

  @Get()
  getHello() {
    return this.health.check([
      () =>
        this.http.pingCheck('google', '/', {
          httpClient: this.httpService,
        }),
    ]);
  }
}
```

@kevjames3 does this work for you? Please let me know so I can release it officially

**Comment by kevjames3** [comment]
Works perfectly!!  I will make a slight update to the example you gave me though.  `httpClient` only allows HttpService, not `HttpService | undefined`.  Casting the variable did the trick :) 

app.controller.ts
```ts

@Controller()
export class AppController {
  constructor(
    private readonly http: HttpHealthIndicator,
    private readonly httpService: HttpService,
    private readonly health: HealthCheckService,
  ) {}

  @Get()
  getHello() {
    return this.health.check([
      () =>
        this.http.pingCheck('google', '/', {
          httpClient: this.httpService as HttpService,
        }),
    ]);
  }
}
```

**Comment by kevjames3** [comment]
Thanks @BrunnerLivio

**Comment by kevjames3** [comment]
@BrunnerLivio - any status update on getting this merged and released?  I would love to leverage it, but am hesitant to use a temp-release npm package in my prod code

**Comment by BrunnerLivio** [comment]
Sorry for the delay. Is released now with [7.2.0](https://github.com/nestjs/terminus/releases/tag/7.2.0) :tada:!
